katello-attach-subscription

```diff
@@ -77,98 +95,239 @@ if @yaml.has_key?(:settings) and @yaml[:settings].is_a?(Hash)
   end
 end
 
+# missing configuration parameters from file or command line are taken from the default one
 @defaults.each do |key,val|
   if not @options.has_key?(key)
     @options[key] = val
   end
 end
 
+# main function
 def vdcupdate()
+  # satellite url has to start with https or PUT will fail with http error
+  unless @options[:uri].start_with?('https://')
+    puts " FATAL ERROR: the uri must start with https://"
+    exit fail
```

Comment:
you want `abort "FATAL ERROR: the uri must start with https://"` instead of puts&exit

also `exit fail` is invalid as `fail` is not defined.